src/DittoMachine.sol

```diff
@@ -550,29 +547,23 @@ contract DittoMachine is ERC721, ERC721TokenReceiver, ERC1155TokenReceiver, Clon
         // no ownership or approval checks cause we're forcing a change of ownership
         if (from != ownerOf[id]) revert FromInvalid();
 
-        unchecked {
-            balanceOf[from]--;
-            balanceOf[to]++;
-        }
-
         ownerOf[id] = to;
 
-        delete getApproved[id];
-        emit Transfer(from, to, id);
+        emit TransferSingle(address(this), from, to, id, 1);
 
         // require statement copied from solmate ERC721 safeTransferFrom()
         require(
             to.code.length == 0 ||
-                ERC721TokenReceiver(to).onERC721Received(msg.sender, from, id, "") ==
-                ERC721TokenReceiver.onERC721Received.selector,
+                ERC1155TokenReceiver(to).onERC1155Received(msg.sender, from, id, 1, "") ==
+                ERC1155TokenReceiver.onERC1155Received.selector,
             "UNSAFE_RECIPIENT"
         );
```

Comment:
Could this require statement be replaced with the `_doSafeTransferAcceptanceCheck` from `1155D.sol`?

Reply:
makes sense. I can make an update.

Reply:
oh wait, The reason I didn't use it was I planned to make `_doSafeTransferAcceptanceCheck` accept calldata type `data` for gas reasons. However, here it would be `memory`.

Reply:
Ah okay, makes sense!